src/doc/tutorials/PythonExpressions.rst

```diff
@@ -0,0 +1,202 @@
+.. _Tutorials_Python_Expressions:
+
+Python Expressions
+==================
+
+.. toctree::
+       :maxdepth: 2
+
+This tutorial describes using Python Expressions.
+
+Python expressions overview
+---------------------------
+
+VisIt_ contains the ability to create expressions using the Python scripting language.
+Python expressions run in the compute engine and run in parallel when VisIt_ is run in parallel.
+Python expressions operate on vtkDataSets, which provide access to variables as well as the mesh coordinates and topology.
+Python expressions also have access to VisIt_'s metadata as well as access to MPI when running in parallel.
+Python expressions return a vtkDataArray, which allows returning new variables.
+It isn't possible to change the mesh topology or coordinates.
+The functionality is available through the *GUI* and the *CLI*.
+When using the *GUI*, they can be created in the *Expressions* window. 
+When using the *CLI*, they can be created with the *DefinePythonExpression* function.
+
+Creating a Python Expression with the GUI
+-----------------------------------------
+
+We'll now go through the steps required to create a Python expression using the GUI.
+
+Let's start by opening a file and creating a plot.
+
+1. Open the file "curv2d.silo".
+2. Create a Psuedocolor plot of *d*.
+
+Now let's go ahead and create the Python expression.
+
+3. Go to *Controls->Expressions*.
+4. This brings up the Expressions window.
+5. Click *New* in the *Expression list* to create a new expression.
+6. Change the *Name* in the *Definition* section to "MyExpression".
+7. Click on the *Python expression editor* tab in the *Definition* section.
+8. Select *Insert variable...->Scalars->d* to add *d* to the *Arguments* text field.
+9. Click *Load script->Template->Simple filter* to add a template of a Python filter in the *Python expression script* editor. 
+
+.. figure:: images/PythonExpression-GUI1.png
+
+   The Expressions window with the simple filter template
+
+At this point you modify the template to create your expression.
+There are a couple of ways in which you can develop your script.
+In either case you would make modifications to the script and then try it out using the Pseudocolor plot.
+Changes to the script can be made either by modifying the script in the *Python expression script* editor or modifying it in an external text editor and then reloading the script.
+Generaly speaking, modifying the script in the *Python expression script* editor is easier than doing it in an external text editor except that it is difficult to tell how many spaces are at the beginning of the line since the editor doesn't use a fixed width font.
```

Comment:
Why don't we use a fixed width font? Can we fix this? Is there already a ticket for that? Given the significance of indentation in python parsing, I am not sure editing in the GUI could ever be *easier* for anything to uncommonly simplistic cases.

Reply:
BTW...wasn't intending to suggest this spacing issue be fixed in this PR. Just that we ought to add an issue for that. I mean if its as simple as causing Qt to use an appropriate font.